pyfr/plugins/tavg.py

```diff
@@ -358,3 +363,196 @@ def __call__(self, intg):
                     self.tstart_acc = intg.tcurr
 
                 self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(BaseCLIPlugin):
+    name = 'tavg'
+
+    @classmethod
+    def add_cli(cls, parser):
+        cmd = cls.name
+        cp = parser.add_parser(cmd, help=f'{cmd} --help')
+        sp = cp.add_subparsers(dest='sub_cmd')
+
+        # Merge command
+        ssp = sp.add_parser('merge', help=f'{cmd} merge --help')
+        ssp.set_defaults(process=cls.merge_cli)
+        ssp.add_argument('solns', nargs='*', help='Averages to merge')
+        ssp.add_argument('output', help='Output file name')
+
+    @cli_external
+    def merge_cli(self):
+        # Open all the solution files
+        self._prepare_files(self.args.solns)
+
+        # Initialise things needed for the merge
+        self._init_tavg_merge()
+
+        with h5py.File(self.args.output, 'w') as outf:
+            # Merge the averages and stds
+            self._merge_data(outf)
+
+            # Merge runtime stats
+            self._merge_runtime_stats()
+
+            # Merge the metadata
+            self._merge_meta(outf)
+
+    def _eval_fun_exprs(self, avars):
+        # Prepare the substitution dictionary
+        subs = {n: avars[:, i] for i, n in enumerate(self.anames)}
```

Comment:
`subs = dict(zip(self.anames, avars.swapaxis(0, 1)))` or something close to that